tachyon/base/buffer/copyable.h

```diff
@@ -162,6 +163,36 @@ class Copyable<std::vector<T>> {
   }
 };
 
+template <typename T>
+class Copyable<std::pmr::vector<T>> {
+ public:
+  static bool WriteTo(const std::pmr::vector<T>& values, Buffer* buffer) {
+    if (!buffer->Write(values.size())) return false;
+    for (const T& value : values) {
+      if (!buffer->Write(value)) return false;
+    }
+    return true;
+  }
+
+  static bool ReadFrom(const ReadOnlyBuffer& buffer,
+                       std::pmr::vector<T>* values) {
+    size_t size;
+    if (!buffer.Read(&size)) return false;
+    values->resize(size);
+    for (T& value : (*values)) {
+      if (!buffer.Read(&value)) return false;
+    }
+    return true;
+  }
+
+  static size_t EstimateSize(const std::pmr::vector<T>& values) {
+    return std::accumulate(values.begin(), values.end(), sizeof(size_t),
+                           [](size_t total, const T& value) {
+                             return total + base::EstimateSize(value);
+                           });
+  }
+};
+
```

Comment:
The code for std::vector and std::pmr::vector is identical for the Copyable aspect. Can we increase code reuse?

Reply:
reflected.